### PR TITLE
To Do list: add & remove

### PR DESCRIPTION
**Pull request by uzairali19**
This branch is to update and delete tasks dynamically. This enables the user to input a task and then displays that instead of hard-coded data.

### CODE REVIEW

File: src/interactive.js [hunk 1]
```diff
@@ -1,75 +1,183 @@
...
+          e.preventDefault();
+          listItem.remove();
+          todoList.splice(i, 1);
+          this.addItem(todoList);
+        });
+      }, 500);
+    });
+  }
+
+  clearAll() {
```
Comment (helman101) [inline src/interactive.js:166]: - [ ] This function should remove all the completed ToDos, currently, your `Clear all completed` button deletes all the ToDos, please fix this.
   - Remember that this function should implement the `Array.filter( )` function 🚀 
   - This function should also update all the indexes

Comment (helman101) [inline src/interactive.js:166]: - [ ] When i press the button that triggers this function, the application stop working, please check and solve this issue 😄

File: src/interactive.js [hunk 2]
```diff
@@ -1,75 +1,183 @@
...
+            todos[index].completed = true;
+            localStorage.setItem('todos', JSON.stringify(todos));
+          } else {
+            divWrapper.classList.remove('checked');
+            todos[index].completed = false;
+            localStorage.setItem('todos', JSON.stringify(todos));
+          }
+        });
+      });
+      this.options();
```
Comment (helman101) [inline src/interactive.js:71]: - [ ] Every time I add a new task, the browser highlights an error in this line of your code, kindly fix this. 🐱
   imagen

File: src/interactive.js [hunk 3]
```diff
@@ -1,75 +1,183 @@
...
+        const listItem = aDiv.parentElement;
+        const p = listItem.children[1];
+
+        todoList[i].todoText = p.innerText;
+        localStorage.setItem('todos', JSON.stringify(todoList));
       });
+    });
+  }
+
+  delItem() {
```
Comment (helman101) [inline src/interactive.js:149]: - [ ] This function is not working properly, sometimes it removes the correct task and sometime it does not, please fix this issue 😄

File: src/interactive.js [hunk 4]
```diff
@@ -1,75 +1,183 @@
...
+    const todoList = JSON.parse(localStorage.getItem('todos'));
+    del.forEach((v, i) => {
+      setTimeout(() => {
+        const aDiv = v.parentElement;
+        const listItem = aDiv.parentElement.parentElement;
+        v.addEventListener('click', (e) => {
+          e.preventDefault();
+          listItem.remove();
+          todoList.splice(i, 1);
+          this.addItem(todoList);
```
Comment (helman101) [inline src/interactive.js:160]: - [ ] I will recommend not using a `setTimeout` in this function, it is not necessary. 🤔 
    - You could erase the task with `Array.filter( )` function, and you remove only the function with the given index

**Review by helman101 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 🔴 

Hi @uzairali19 👋, 

Nice job so far with this project 😄, but there are still some changes to do.

## Required Changes ♻️

- [x] _Check the comments under the review._

Cheers and Happy coding!👏👏👏

gif

If anything is not 100% clear please leave your questions or comments in the PR thread, we will be glad to help you 😄.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/interactive.js [hunk 5]
```diff
@@ -1,75 +1,187 @@
...
+        });
+      });
     }
+  }
 
-    const check = document.querySelectorAll('.check-input');
-    const todos = JSON.parse(localStorage.getItem('todos'));
-    check.forEach((check, index) => {
-      check.addEventListener('change', (e) => {
+  editItem() {
```
Comment (SajjadAhmad14) [inline src/interactive.js:132]: [ OPTIONAL ] This function does not work as the user is unable to edit the description of the item. Please fix the problem.

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Uzair Ali, 
## Congratulations 🎉 Thank you for implementing changes suggested previously 🙏

## Happy Coding 👏👏👏
#### Please leave any questions or comments in the PR if you have any doubts or unclarity!
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/interactive.js (hunk 5 above)
Comment (uzairali19) [reply src/interactive.js:132]: It works if you click the edit button after editing the task description.

**Pull request merged**